v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstovectorsearch/BigtableChangeStreamsToVectorSearch.java

```diff
@@ -0,0 +1,235 @@
+      "bigtableRpcTimeoutMs"
+    },
+    // TODO(meagar): Documentation link
+    documentation =
+        "https://cloud.google.com/dataflow/docs/guides/templates/provided/cloud-bigtable-change-streams-to-vector-search",
+    flexContainerName = "bigtable-changestreams-to-vector-search",
+    contactInformation = "https://cloud.google.com/support",
+    streaming = true)
+public final class BigtableChangeStreamsToVectorSearch {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(BigtableChangeStreamsToVectorSearch.class);
+
+  /**
+   * Main entry point for executing the pipeline.
+   *
+   * @param args The command-line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    LOG.info(
+        "Starting to replicate change records from Cloud Bigtable change streams to Vector Search");
+
+    BigtableChangeStreamsToVectorSearchOptions options =
+        PipelineOptionsFactory.fromArgs(args)
+            .withValidation()
+            .as(BigtableChangeStreamsToVectorSearchOptions.class);
+
+    // TODO(meagar): Copied from Ron's template, when is this required?
+    // Wait for pipeline to finish only if it is not constructing a template.
+    // if (options.as(DataflowPipelineOptions.class).getTemplateLocation() == null) {
+    //   result.waitUntilFinish();
+    // }
+    try {
+      run(options);
+    } catch (IOException e) {
+      LOG.error("{}", e);
+    }
+
+    LOG.info("Completed pipeline setup");
+  }
+
+  public static PipelineResult run(BigtableChangeStreamsToVectorSearchOptions options)
+      throws IOException {
+    options.setStreaming(true);
+    options.setEnableStreamingEngine(true);
+
+    // TODO(meagar): Copied from Ron's template, is this important?
+    // Do not validate input fields if it is running as a template.
+    // if (options.as(DataflowPipelineOptions.class).getTemplateLocation() != null) {
+    //   read = read.withoutValidation();
+    // }
+
+    Instant startTimestamp =
+        options.getBigtableChangeStreamStartTimestamp().isEmpty()
+            ? Instant.now()
+            : toInstant(Timestamp.parseTimestamp(options.getBigtableChangeStreamStartTimestamp()));
+
+    String bigtableProjectId = getBigtableProjectId(options);
+
+    LOG.debug("  - startTimestamp {}", startTimestamp);
+    LOG.debug("  - bigtableReadInstanceId {}", options.getBigtableReadInstanceId());
+    LOG.debug("  - bigtableReadTableId {}", options.getBigtableReadTableId());
+    LOG.debug("  - bigtableChangeStreamAppProfile {}", options.getBigtableChangeStreamAppProfile());
+    LOG.debug("  - embeddingColumn {}", options.getEmbeddingColumn());
+    LOG.debug("  - crowdingTagColumn {}", options.getCrowdingTagColumn());
+    LOG.debug("  - project {}", options.getProject());
+
+    String indexName = options.getVectorSearchIndex();
+    // TODO(meagar): Raises, do we handle this, or allow exceptions to escape?
+    String vertexRegion = Utils.extractRegionFromIndexName(indexName);
```

Comment:
Should I allow exceptions to bubble out of this method, or capture them and emit some kind of error?

Reply:
I think this is fine. Generally speaking we want to halt execution of the pipeline the moment we encounter something like this (i.e. invalid index name)